File: Ejercicios/08.Perrin/Perrin.py
def main():
    p0 = 3
    p1 = 0
    p2 = 2
    
    n = int(input("ingrese n:"))
    if n == 0:
        print(f"({p0})")
    elif n == 1:
        print(f"({p1})")
    elif n == 2:
        print (f"({p2})")
    else:
        perrin = 0
        for i in range (3, n + 1):
            perrin = p1 + p0
            p0 = p1
            p1 = p2
            p2 = perrin
        print(f"({perrin})")

if __name__ == "__main__":
    main()

File: Ejercicios/08.Perrin/Perrin.c
#include <stdio.h>
#include <stdlib.h>

int main(){
    int n, i, p0=3, p1=0, p2=2, perrin;
    do{
        printf("Ingrese n:");
        scanf("%d",&n);
        if (n < 0)
            printf("Error debe ser >=0"); 
    } while (n < 0);

    if (n == 0)
        printf("(%d)", p0);
    else if (n == 1)
        printf("(%d)", p1);
    else if (n == 2)
        printf("(%d)", p2);
    else{
        if (n == 2)
            printf("(%d)", p2);
        
        for (i = 3; i <= n; i++){
            perrin = p1 + p0;

            p0 = p1;
            p1 = p2;
            p2 = perrin;
        }
        
        printf("(%d)", perrin);  
    }

    return EXIT_SUCCESS;
}